mesonbuild/backend/backends.py

```diff
@@ -594,6 +595,22 @@ def get_executable_serialisation(
             exe_wrapper = None
 
         workdir = workdir or self.environment.get_build_dir()
+
+        # Must include separators as well
+        needs_rsp_file = can_use_rsp_file and sum(len(i) for i in cmd_args) + (len(cmd_args) - 1) >= get_rsp_threshold()
```

Comment:
There is also the length of the command itself that is missing here. But otoh, `get_rsp_threshold()` returns a conservative estimate, so the length doesn't need to be excessively precise.

Maybe would it be simpler to do something similar to what I did in i18n.py:230:

`sum(len(i) + 1 for i in cmd_args)`